databricks/koalas/indexes.py

```diff
@@ -1632,6 +1632,55 @@ def repeat(self, repeats: int) -> "Index":
         else:
             return ks.concat([kdf] * repeats).index
 
+    def asof(self, label):
+        """
+        Return the label from the index, or, if not present, the previous one.
+
+        Assuming that the index is sorted, return the passed index label if it
+        is in the index, or return the previous index label if the passed one
+        is not in the index.
+
```

Comment:
Can you add a note that this API is dependent on `is_monotonic_increasing` which can be expensive? Also, seems we should update the note in the docstring at `truncate`.

Reply:
Yes, then it is okay that I'll modify both in this PR?